src/storage-client/src/controller.rs

```diff
@@ -1486,8 +1486,9 @@ where
 /// A wrapper struct that presents the adapter token to a format that is understandable by persist
 /// and also allows us to differentiate between a token being present versus being set for the
 /// first time.
+// TODO(aljoscha): Move this to a different module?
```

Comment:
we can make this `pub(crate)` again once only the controller holds critical handles. I can change the TODO to reflect that.